main.go

```diff
@@ -47,36 +47,54 @@ func main() {
 		os.Exit(1)
 	}
 
-	testLogger := ldtest.ConsoleTestLogger{
+	var testLogger ldtest.TestLogger
+	consoleLogger := ldtest.ConsoleTestLogger{
 		DebugOutputOnFailure: params.debug || params.debugAll,
 		DebugOutputOnSuccess: params.debugAll,
 	}
+	if params.jUnitFile == "" {
+		testLogger = consoleLogger
+	} else {
+		testLogger = &ldtest.MultiTestLogger{Loggers: []ldtest.TestLogger{
+			consoleLogger,
+			ldtest.NewJUnitTestLogger(params.jUnitFile, harness.TestServiceInfo(), params.filters),
+		}}
+	}
 
-	var results ldtest.Results
-	capabilities := harness.TestServiceInfo().Capabilities
+	enabledCapabilities := harness.TestServiceInfo().Capabilities
+	var allCapabilities framework.Capabilities
 	switch {
-	case capabilities.Has(servicedef.CapabilityServerSide):
+	case enabledCapabilities.Has(servicedef.CapabilityServerSide):
 		fmt.Println("Running server-side SDK test suite")
-		results = sdktests.RunServerSideTestSuite(harness, params.filters.Match, testLogger)
-	case capabilities.Has(servicedef.CapabilityClientSide):
+		allCapabilities = sdktests.AllImportantServerSideCapabilities()
```

Comment:
I added the line about `allCapabilities` to fix a desired behavior that wasn't happening: it is supposed to tell you at the top of the test output whether tests will be skipped because of a missing capability. This requires comparing the list of capabilities supported by the test service to a list of all the ones we would like them to implement.